pkg/base1/test-location.js

```diff
@@ -298,4 +298,22 @@ QUnit.test("test", function (assert) {
     window.location.hash = "#/other";
 });
 
+/* Set url_root in beforeEach so the test does not flake */
+QUnit.module("url_root tests", {
+    beforeEach: () => {
+        window.mock = { url_root: "cockpit" };
+    },
+    afterEach: () => {
+        window.mock = null;
+    }
+});
+
+QUnit.test("encode", function (assert) {
+    let path = cockpit.location.encode("path", "", true);
+    assert.deepEqual(path, "/cockpit/path", "path is correct");
```

Comment:
A string isn't very deep, is it? :grin:  (Should work fine that way, of course).